### PR TITLE
Convert to range-for loops

### PR DESCRIPTION
**Pull request by lgritz**
Suggestions courtesy of clang-tidy.

### CODE REVIEW

File: src/field3d.imageio/field3doutput.cpp [hunk 1]
```diff
@@ -278,8 +278,8 @@ Field3DOutput::put_parameter (const std::string &name, TypeDesc type,
                 Strutil::split (format_list, format_prefixes, ",");
                 format_prefixes_initialized = true;
             }
-            for (size_t i = 0, e = format_prefixes.size();  i < e;  ++i)
-                if (Strutil::iequals (prefix, format_prefixes[i]))
+            for (const auto & f : format_prefixes)
```
Comment (fpsunflower) [inline src/field3d.imageio/field3doutput.cpp]: I prefer `const auto&` - did `clang-tidy` suggest the spacing?

**Comment by fpsunflower** [comment]
Other than the whitespace nit pick -- LGTM!

Very cool to see how many cases it can recognize.

File: src/field3d.imageio/field3doutput.cpp (hunk 1 above)
Comment (lgritz) [inline src/field3d.imageio/field3doutput.cpp]: Yes. In most cases, I just let it be. Though in cases where I chose to go and correct its choice of variable names, I also "fixed" the spacing.

If you feel strongly about it, I can fix them all. But I was prepared to just accept the free work that clang-tidy did for me and not nit pick a slightly different, but equally readable, spacing than I would have used if I'd made the changes by hand.

**Comment by lgritz** [comment]
clang-tidy did a pretty good job. Much of the time, it picks good variable names for you. For example, if you have a loop like

    for (size_t i = 0; i < values.size(); ++i)
        sum += values[i];

it is smart enough to see that the array you are iterating, "values", ends in "s", so it makes it singular like this:

    for (auto & value : values)
        sum += value;

Neat!

But there are some cases where it screws it up:

    for (size_t i = 0; i < meshes.size(); ++i)

will turn into

    for (auto & meshe : meshes)

Oops.  So there were a few cases like that where I went in and edited by hand to pick better names, and in those cases I also fixed any discrepancies with spacing around the `&`.

File: src/field3d.imageio/field3doutput.cpp (hunk 1 above)
Comment (fpsunflower) [inline src/field3d.imageio/field3doutput.cpp]: I think its worth making them all consistent.

**Comment by lgritz** [comment]
Updated with whitespace changes.

**Comment by fpsunflower** [comment]
LGTM! Thanks :)

**Pull request merged**